### ISSUE TITLE
The plugin doesn't scale

### ISSUE THREAD

**Issue by Bryksin** [jenkinsci/amazon-ecs-plugin#111]
Hi

**Infrastructure:**
I created the ECS cluster for slave agents with Auto Scaling based on CPU value. 
The initial instance was taken with 2 Cores and 4 Ram, where for build task I configured to consume 1 CPU and 2GB Ram per build
Therefore, I would imagine that 2 agents can be spawned per instance and whenever 2nd agent will kick off the build, the CPU value for AutoScailing will go above the threshold and new instance will be spawned, what would allow deploying more agents to ECS.

**Jenkins Project**
I configured the plugin, point to the ECS cluster described above and used the base Docker image (btw look at: #110 ) with 1024 per CPU and 2048 RAM as already described above

I created 2 dummy projects with parallel builds allowed and with a single build step `sleep for 5 min`
After I executed 3 builds per each project = 6 builds in total pending in the Q

**Expected**
The plugin will start spawning as many agents as possible to accommodate all builds in the Q, to accommodate parallel builds. 
And infrastructure behind ECS cluster will scale automatically based on the CPU load (it base AWS AutoScale functionality, outside of this plugin context) 

**Actual Result**
Only 1 agent was spawned, where rest of the build tasks remains in the Q waiting for that single agent instead of spawning new agents for parallel execution

What did I configure wrong?

**Comment by bradq** [comment]
@Bryksin It would be more professional to the open source developers giving their time to this project, and more helpful to your own work, if you appreciate that perhaps you've misconfigured the plugin. It's been around for years, and many of us use it to scale with great effect. 

I'm not affiliated in any way with the project, but would suggest you look over the documentation, and logs, to figure out what's going on. I've found [this particular thread](https://www.reddit.com/r/devops/comments/9rpfbq/lets_talk_bestpractice_jenkins_on_aws_ecs/) to be rather helpful. 

There are often issues with quick spawning of similar containers in Jenkins, something that has everything to do with how Jenkins internally processes queuing. I've found it quite helpful to use the model of:

```
agent {
   ecs {
     inheritFrom 'jnlp-docker'
   }
}
```

rather than 

```
agent {
    label 'jnlp-docker'
}
```

While the "ECS" type is intended to allow override of base templates, it does have the side effect of creating new labels in Jenkins that often fixes this type of problem. 

I would also suggest that you simply take the time to submit a PR for #110 rather than attempting to bump it 6 hours after submitting it. It's a simple change, and a good way to give back to the project that others have spent years working on.

**Comment by markyjackson-taulia** [comment]
@Bryksin We gladly accept PR's, feature requests and positive feedback for our spare time efforts.
Perhaps I misunderstood your titled (and if that is the case I apologize).

**Comment by Bryksin** [comment]
@bradq thank you very much for the clarification of my misconfiguration, I was using quick UI steps to configure dummy projects rather than Jenkins file, not sure if such fine-tuning is available there.

I also promise to fix #110 in the nearest future when I will find spare 10 min :)

Thank you one more time for clarification, I would imagine this issue can be closed

**Comment by markyjackson-taulia** [comment]
Fix for #110 is in already :-)
